src/crimson/os/seastore/onode_manager/staged-fltree/tree.h

```diff
@@ -265,10 +265,10 @@ class Btree {
     value_config_t vconf{value_builder.get_header_magic(), _vconf.payload_size};
     return seastar::do_with(
       full_key_t<KeyT::HOBJ>(obj),
-      [this, &t, vconf](auto& key) -> eagain_ifuture<std::pair<Cursor, bool>> {
+      [this, &t, vconf, &obj](auto& key) -> eagain_ifuture<std::pair<Cursor, bool>> {
         ceph_assert(key.is_valid());
-        return get_root(t).si_then([this, &t, &key, vconf](auto root) {
-          return root->insert(get_context(t), key, vconf, std::move(root));
+        return get_root(t).si_then([this, &t, &key, vconf, &obj](auto root) {
```

Comment:
Ditto, `obj` is not supposed to have extented lifecycle beyond async operations.